agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4398,10 +4281,210 @@ bool slave_thread::send_cmdu_to_controller(ieee1905_1::CmduMessageTx &cmdu_tx)
                                   backhaul_params.bridge_mac);
 }
 
-void slave_thread::handle_channel_preference_query_message(Socket *sd,
-                                                           ieee1905_1::CmduMessageRx &cmdu_rx)
+bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    // Check if this is a M1 message that we sent to the controller, which was just looped back.
+    // The M1 and M2 messages are both of CMDU type AP_Autoconfiguration_WSC. Thus,
+    // when we send the M2 to the local agent, it will be published back on the local bus because
+    // the destination is our AL-MAC, and the controller does listen to this CMDU.
+    // Ideally, we should use the message type attribute from the WSC payload to distinguish.
+    // Unfortunately, that is a bit complicated with the current tlv parser. However, there is another
+    // way to distinguish them: the M1 message has the AP_Radio_Basic_Capabilities TLV,
+    // while the M2 has the AP_Radio_Identifier TLV.
+    // If this is a looped back M2 CMDU, we can treat is as handled successfully.
+    if (cmdu_rx.getNextTlvType() == wfa_map::TLV_AP_RADIO_BASIC_CAPABILITIES)
+        return true;
+
+    /**
+    * @brief Parse AP-Autoconfiguration WSC which should include one AP Radio Identifier
+    * TLV and one or more WSC TLV containing M2
+    */
+    auto ruid = cmdu_rx.addClass<wfa_map::tlvApRadioIdentifier>();
+    if (ruid == nullptr) {
+        LOG(ERROR) << "Failed to get tlvApRadioIdentifier TLV";
+        return false;
+    }
+
+    // Check if the message is for this radio agent by comparing the ruid
+    if (!config.radio_identifier.compare(network_utils::mac_to_string(ruid->radio_uid())))
+        return true;
+
+    LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_WIFI_SIMPLE_CONFIGURATION_MESSAGE";
```

Comment:
Already said that before - the message should be called WSC

Reply:
No, this message is autoconfig which has a WSC TLV.

Reply:
`AP_AUTOCONFIGURATION_WSC_MESSAGE`

Reply:
ok, I'll do that in a separate commit